snapcraft/commands/remote.py

```diff
@@ -93,15 +93,15 @@ def fill_parser(self, parser: argparse.ArgumentParser) -> None:
         )
         parser.add_argument(
             "--build-on",
+            type=lambda arg: [arch.strip() for arch in arg.split(",")],
             metavar="arch",
-            nargs="+",
             help=HIDDEN,
         )
         parser.add_argument(
             "--build-for",
+            type=lambda arg: [arch.strip() for arch in arg.split(",")],
```

Comment:
Does this support something wacky like `--build-for=amd64 --build-for=arm64,armhf`?

Reply:
Good question.  When there are duplicates, argparse/craft-cli uses the last `--build-{on|for}`, silently ignores the other `--build-{on|for}`s, and always provides a single string for this lambda.

Normally I wouldn't add a test for standard behavior, but since this is a delicate situation, I added an extra unit test.

Reply:
to be clear, this wouldn't be a change in behavior right? The previous code would also only use the last --build-x?

Reply:
Correct, this behavior is unchanged.

The only "loss" is that `snapcraft remote-build --build-for amd64 arm64` doesn't work.  

And I think that "loss" is OK because it is a regression in the new remote builder: snapcraft 7 only supported comma-separated architectures, not space-separated architectures.